crates/rpc/rpc-eth-api/src/helpers/estimate.rs

```diff
@@ -330,32 +327,11 @@ pub fn update_estimated_gas_range(
             // Cap the highest gas limit with the succeeding gas limit.
             *highest_gas_limit = tx_gas_limit;
         }
-        ExecutionResult::Revert { .. } => {
-            // Increase the lowest gas limit.
+        ExecutionResult::Revert { .. } | ExecutionResult::Halt { .. } => {
+            // We know that transaction succeeded with a higher gas limit before, so any failure
+            // means that we need to increase it.
```

Comment:
do we want to add some trace level logging here right away to differentiate the two in debugging? does the migration to alloy not change so much code the we can still consider the sigp audit of the reth code to apply to the code in alloy?